charts/victoria-metrics-k8s-stack/templates/_helpers.tpl

```diff
@@ -0,0 +1,126 @@
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "victoria-metrics-k8s-stack.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create }}
+{{- default (include "victoria-metrics-k8s-stack.fullname" .) .Values.serviceAccount.name }}
+{{- else }}
+{{- default "default" .Values.serviceAccount.name }}
+{{- end }}
+{{- end }}
+
+{{/*
+Create the name for VMSingle
+*/}}
+{{- define "victoria-metrics-k8s-stack.vmsingleName" -}}
+{{- .Values.vmsingle.name | default (printf "vmsingle-%s" (include "victoria-metrics-k8s-stack.fullname" .))}}
+{{- end }}
+
+
+{{/*
+Common labels
+*/}}
+{{- define "victoria-metrics-k8s-stack.labels" -}}
+helm.sh/chart: {{ include "victoria-metrics-k8s-stack.chart" . }}
+{{ include "victoria-metrics-k8s-stack.selectorLabels" . }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end }}
+
+{{/*
+Selector labels
+*/}}
+{{- define "victoria-metrics-k8s-stack.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "victoria-metrics-k8s-stack.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
+
+{{/*
+VM endpoint
+*/}}
+{{- define "victoria-metrics-k8s-stack.vmEndpoint" -}}
+{{- if .Values.vmsingle.enabled -}}
+url: "http://{{ include "victoria-metrics-k8s-stack.vmsingleName" .}}.{{ .Release.Namespace }}.svc:{{ .Values.vmsingle.spec.port | default 8429 }}"
+{{- end }}
+{{- end }}
+
+
+{{/*
+Alermanager config secret name
+*/}}
+{{- define "victoria-metrics-k8s-stack.alertmanagerConfigSecret" -}}
+{{- if .Values.alertmanager.secretName -}}
+configSecret: {{ .Values.alertmanager.secretName | default (printf "%s-alertmanager" (include "victoria-metrics-k8s-stack.fullname" .)) }}
+{{- else -}}
+configSecret: {{ .Values.alertmanager.spec.configSecret | default (printf "%s-alertmanager" (include "victoria-metrics-k8s-stack.fullname" .)) }}
+{{- end }}
+{{- end }}
+
+{{/*
+Alermanager spec
+*/}}
+{{- define "victoria-metrics-k8s-stack.alertmanagerSpec" -}}
+{{ deepCopy (include "victoria-metrics-k8s-stack.alertmanagerConfigSecret" . | fromYaml ) | mergeOverwrite ( .Values.alertmanager.spec ) | toYaml }}
+{{- end }}
+
+{{/*
+VMAlert spec
+*/}}
+{{- define "victoria-metrics-k8s-stack.vmAlertSpec" -}}
+{{- $vmAlertStackRemoteWrite := dict "remoteWrite" ( include "victoria-metrics-k8s-stack.vmEndpoint" . | fromYaml ) -}}
+{{- $vmAlertStackRemoteRead := dict "remoteRead" ( include "victoria-metrics-k8s-stack.vmEndpoint" . | fromYaml ) -}}
+{{- $vmAlertStackDatasource := dict "datasource" ( include "victoria-metrics-k8s-stack.vmEndpoint" . | fromYaml ) -}}
+{{- $vmAlertStackNotifier := dict "notifier"  ( dict "url" ( printf "http://vmalertmanager-%s.%s.svc:9093" (include "victoria-metrics-k8s-stack.fullname" .) .Release.Namespace ) ) -}}
```

Comment:
i suggest to use notifiers instead of notifier and build per pod notifier url according to replica size of alertmanager.

Reply:
lets have this in separate PR, to be able also to test it separatelly

Reply:
agreed